cli/params.yaml

```diff
@@ -0,0 +1,34 @@
+params:
```

Comment:
Is my understanding correct that all of the data here would be added as metadata to a model in the model registry? I wonder if the pipeline should be able to figure out some of the information based on standard model information in the registry, rather than having metadata for each pipeline parameter. :thinking: 

Having not thought about it too much, I wonder if the _output_ of the MLOps Pipeline (image reference) is more relevant information to store with a model, rather than the pipeline _input_ params?

Reply:
> Is my understanding correct that all of the data here would be added as metadata to a model in the model registry? I wonder if the pipeline should be able to figure out some of the information based on standard model information in the registry, rather than having metadata for each pipeline parameter. 🤔

- The implemented CLI use case stores the metadata once and then reuses it whenever an image build is triggered. 
- I believe some metadata can be figured out by the pipeline, and some need to be stored somewhere. E.g. the location of the model.
- I see it as inevitable that such metadata shouldn't be stored in the current naive format of arbitrary key-value pairs. Instead, the MR should have some properties that have a semantic meaning that the pipeline and/or CLI can use. However, the current approach seems okay for proofing the concept/demo purposes.

> 
> Having not thought about it too much, I wonder if the _output_ of the MLOps Pipeline (image reference) is more relevant information to store with a model, rather than the pipeline _input_ params?

Well it depends on where we're gonna end up storing the model container image information, cause one option that's up for future investigation is to use the container registry and [OCI 1.1 to manage the artifacts metadata](https://github.com/opencontainers/image-spec/blob/main/manifest.md#guidelines-for-artifact-usage)  and in such case I can imagine a lot of the required metadata can be available in the container registry or have to be updated by the pipeline at some stage.

Currently, it'd definitely make sense to make the pipeline update the status of the model artifact and version in the model registry with the pipeline build result and the container image sha etc.